### PR TITLE
Prevent duplicate invitations

### PR DESCRIPTION
**Pull request by gazayas**
Context: https://discord.com/channels/836637622432170028/1130331556137947188

Tests at the joint PR
- ~https://github.com/bullet-train-co/bullet_train/pull/812~
- https://github.com/bullet-train-co/bullet_train/pull/856

As you can see from the tests, we prevent sending invitations to the same email, but we can still use the resend button to send another invitation to the same email.

### CODE REVIEW

**Review by vfonic — CHANGES_REQUESTED** [review]
This fixes the underlying issue only for this particular controller action.

The underlying issue is: user can have only one membership to the same team

I found another way to "exploit" this, without having to write any new code.*

Steps:

1. Send invite to michael@gmail.com (asd@ada-dev.com in screenshot below)
2. Accept invite as michael@gmail.com (asd@ada-dev.com)
3. Send invite to michael@our-big-corp.com (desi@ada-dev.com)
4. Accept invite for michael@our-big-corp.com (desi@ada-dev.com) as michael@gmail.com (asd@ada-dev.com)

Expected:

You see an error saying that you're already a member

Actual:

You now have two memberships connected to the same `User`.

Screenshot 2023-07-18 at 10 39 50

This could create permission bugs if we're only checking "first" (which comes) membership (`Membership.find_by(user:, team:)` or similar).


*What I mean by "without having to write any new code" is related to my comment on Discord:

> I just tested it out in rails console:

```ruby
m = Membership.all.sample
newm = m.dup
newm.save # creates new Membership record witht the same [user_id, team_id]
```

> Maybe this issue/bug never happened, but there's nothing preventing it from happening. It could be as simple code as above or it could be a code where someone could have a data migration and/or task assigning memberships to teams or someone might have double clicked on submit button because of slow connection (wouldn't that create two memberships)?

> I dug deeper and it seems like you can create as many invitations as you want for: the same email, the same team, the same role/permissions. These are all the same email addresses in the screenshot. This definitely looks like a bug to me.

**Comment by jagthedrummer** [comment]
It does seem like there should be database constraints that ensure unique `team_id`/`user_id` combinations on the `memberships` table. And maybe also for the `team_id`/`user_email` combo.

**Comment by jagthedrummer** [comment]
@gazayas, just curious what you think about the feedback offered above. Would a database constraint be a bad idea? And if we did add one, would it mean that these changes could be simpler?

**Comment by gazayas** [comment]
@jagthedrummer Definitely not a bad idea, will have to take a second look

**Comment by gazayas** [comment]
@jagthedrummer I changed some things around to make sure duplicate invitations aren't sent, does this go along the lines of what you were thinking of? Would be glad to to take a closer look if not.

I opened a new joint PR for this one too:
- https://github.com/bullet-train-co/bullet_train/pull/856

**Comment by jagthedrummer** [comment]
@gazayas, this is looking great! Gonna merge it and release it today in `1.3.12`.

File: bullet_train/config/locales/en/invitations.en.yml [hunk 1]
```diff
@@ -59,6 +59,7 @@ en:
       resent: 'Invitation was successfully resent.'
       resent_error: Sorry, we couldn't find an invitation to resend.
       doesnt_exist: Sorry, but we couldn't find your invitation. Please contact an administrator on the team to have them send you another one.
+      duplicate: 'Sorry, a member with the email %{user_email} has already been invited.'
```
[the remark this answers is not in the document]
Comment (gazayas) [reply bullet_train/config/locales/en/invitations.en.yml:62]: Opened #473 to resolve this.

**Pull request merged**